plugins/cli/src/lib.rs

```diff
@@ -28,8 +28,8 @@ pub use parser::{ArgData, Matches, SubcommandMatches};
 pub struct Cli<R: Runtime>(PluginApi<R, Config>);
 
 impl<R: Runtime> Cli<R> {
-    pub fn matches(&self) -> Result<parser::Matches> {
-        parser::get_matches(self.0.config(), self.0.app().package_info())
+    pub fn matches(&self, args: Option<Vec<String>>) -> Result<parser::Matches> {
+        parser::get_matches(self.0.config(), self.0.app().package_info(), args)
     }
```

Comment:
Since this is a breaking change, let's add another function `matches_from` here

```suggestion
    pub fn matches(&self) -> Result<parser::Matches> {
        parser::get_matches(self.0.config(), self.0.app().package_info(), None)
    }

     pub fn matches_from(&self, args: Vec<String>) -> Result<parser::Matches> {
        parser::get_matches(self.0.config(), self.0.app().package_info(), Some(args))
    }
```